src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -782,18 +716,19 @@ int D4NFilterObject::D4NFilterReadOp::D4NFilterGetCB::handle_data(bufferlist& bl
       std::string oid = this->oid + "_" + std::to_string(ofs) + "_" + std::to_string(bl_len);
       ret = filter->get_cache_driver()->put_async(save_dpp, oid, bl, bl.length(), source->get_attrs());
       if (ret == 0) {
-        block.size = bl.length();
-        block.blockId = oid;
-        block.hostsList.push_back(blockDir->get_addr().host + ":" + std::to_string(blockDir->get_addr().port)); //check if this is the correct way to get the local addr
-        block.size = source->get_obj_size();
-        block.cacheObj.bucketName = source->get_bucket()->get_name();
+        block.version = oid;
+        block.size = bl.length(); // TODO: fill out block entirely
+        block.size = source->get_obj_size(); // which is correct? -Sam
+        block.hostsList.push_back("127.0.0.1:6379" /*current cache addr*/); 
         block.cacheObj.objName = source->get_key().get_oid();
-        if (filter->get_policy_driver()->cachePolicy->get_block(save_dpp, &block, filter->get_cache_driver()) == 0) {
-          filter->get_policy_driver()->cachePolicy->insert(save_dpp, oid, ofs, bl.length(), filter->get_cache_driver());
+        block.cacheObj.bucketName = source->get_bucket()->get_name();
+        block.cacheObj.hostsList.push_back("127.0.0.1:6379" /*current cache addr*/);
+        if (filter->get_policy_driver()->get_cache_policy()->get_block(save_dpp, &block, filter->get_cache_driver(), *save_y) == 0) {
+          filter->get_policy_driver()->get_cache_policy()->insert(save_dpp, oid, ofs, bl.length(), "", filter->get_cache_driver(), *save_y);
```

Comment:
As noted above, the 'insert' will move out of here and to the cache driver.

Reply:
also, get_block() must be called before put_async() to free up space in the cache node.
The order of calls must be:
1. get_block()
2. put_async()
3. insert()

I am going to correct this in the topic branch.